Public/Src/FrontEnd/Nuget/WorkspaceNugetModuleResolver.cs

```diff
@@ -918,16 +922,52 @@ private bool CanReuseSpecFromDisk(NugetAnalyzedPackage analyzedPackage)
         {
             var packageDsc = GetPackageDscFile(analyzedPackage).ToString(PathTable);
 
-            if (analyzedPackage.Source == PackageSource.Disk &&
-                analyzedPackage.PackageOnDisk.PackageDownloadResult.SpecsFormatIsUpToDate &&
-                File.Exists(packageDsc))
+            var state = ReadPackageStateFile(packageDsc + SpecGenerationVersionFileSuffix);
+            if (state.fileFormat == SpecGenerationFormatVersion &&
+                state.fingerprint != null &&
+                state.fingerprint == analyzedPackage.PackageOnDisk.PackageDownloadResult.FingerprintHash && 
+                File.Exists(packageDsc) &&
+                File.Exists(GetPackageConfigDscFile(analyzedPackage).ToString(PathTable)))
             {
                 return true;
             }
 
             return false;
         }
 
+        private static (int fileFormat, string fingerprint) ReadPackageStateFile(string path)
+        {
+            if(File.Exists(path))
+            {
+                int fileFormat;
+                string[] lines = File.ReadAllLines(path);
+                if(lines.Length == 2)
+                {
+                    if (int.TryParse(lines[0], out fileFormat))
+                    {
+                        string fingerprint = lines[1];
+                        return (fileFormat, fingerprint);
+                    }
+                }
+            }
+
+            // Error
+            return (-1, null);
+        }
+
+        private static void WritePackageStateFile(string path, (int fileFormat, string fingerprint) data)
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(path));

```

Comment:
do we need try/catch with a warning that the next build the package will be downloaded again? Not sure if the old code, or my attempt and rewrite 18 months ago had that...

Reply:
added